awscli/customizations/s3/filegenerator.py

```diff
@@ -208,13 +207,19 @@ def list_files(self, path, dir_op):
                         for x in self.list_files(file_path, dir_op):
                             yield x
                     else:
-                        size, last_update = get_file_stat(file_path)
-                        last_update = self._validate_update_time(
-                            last_update, path)
-                        yield (
-                            file_path,
-                            {'Size': size, 'LastModified': last_update}
-                        )
+                        stats = self._safely_get_file_stats(path, file_name=name)
```

Comment:
Sorry. I am still a little confused on how it is different.

So on line 199: https://github.com/aws/aws-cli/pull/2333/files#diff-2d6153b2f8310981a59f57a017f7a2a3R199 there is:
``file_path = join(path, name)``

Then on line 215: https://github.com/aws/aws-cli/pull/2333/files#diff-2d6153b2f8310981a59f57a017f7a2a3R215. The logic is:
``file_path = path if file_name is None else os.path.join(path, file_name)``

So then if you make line 210: https://github.com/aws/aws-cli/pull/2333/files#diff-2d6153b2f8310981a59f57a017f7a2a3R210:
`` stats = self._safely_get_file_stats(file_path)``

It would produce the same result as the original line you are adding because the variable ``file_path`` on line 199 would be the same result as on line 215 with ``os.path.join(path, file_name)``.